CONTRIBUTING.md

```diff
@@ -29,3 +29,138 @@ Recommendations and requirements for how to best contribute to Parquet. We striv
+   used as dependency in parquet-java.
+
+### Releases
+
+The Parquet PMC aims to do releases of the format package only as needed when
+new features are introduced. If multiple new features are being proposed
+simultaneously some features might be consolidated into the same release.
+Guidance is provided below on when implementations should enable features added
+to the specification.  Due to confusion in the past over parquet versioning it
+is not expected that there will be a 3.x release of the specification in the
+foreseeable future.
+
+### Compatibility and Feature Enablement
+
+For the purposes of this discussion we classify features into the following buckets:
+
+1. Backwards compatible. A file written under an older version of the format
+   should be readable under a newer version of the format.
+
+2. Forwards compatible. A file written under a newer version of the format with
+   the enabled feature can be read under an older version of the format, but
+   some information might be missing or performance might be suboptimal.
+
+3. Forward incompatible. A file written under a new version of the format with
+   the feature enabled cannot be read under an older version of the format (e.g.
+   Adding a new compression algorithm).
+
+New features are intended to be widely beneficial to users of Parquet, and
+therefore it is hoped third-party implementations will adopt them quickly after
+they are introduced. It is assumed that writing new parts of the format, and
+especially forward incompatible features, will be configured with feature flag
+defaulted to "off" and at some future point the features are turned on by default
+(reading of the new feature will typically be enabled without configuration or
+defaulted to on). Some amount of lead time is desirable to ensure a critical
+mass of Parquet implementations support a feature to avoid compatability issues
+across the ecosystem.  Therefore, the Parquet PMC gives the following
+recommendations for managing features:
+
+1. Backwards compatibility is the concern of implementations but given the
+   ubiquity of Parquet and the length of time it has been used, libraries should
+   support reading older version of the format to the greatest extent possible.
+
+2. Forward compatible features/changes may be used by default in implementations
+   once the parquet-format containing those changes has been formally released.
+   For features that may pose a significant performance regression to older
+   format readers, libaries should consider delaying default enablement until 1
+   year after the release of the parquet-java implementation that contains the
+   feature implementation.
+
+3. Forwards incompatible features/changes should not be turned on by default
+   until 2 years after the parquet-java implementation containing the feature is
+   released. It is recommended that changing the default value for a forward
+   incompatible feature flag be done as part of a major release of an
```

Comment:
Given that "major release" is vague and potentially misleading, I would remove this recommendation as it is not really helpful for implementors, IMHO.

Reply:
I'll see if I can rephrase to something more reasonable otherwise we can remove it.

Reply:
I tried to rephrase by only giving examples, let me know if you think this language is still too strong or still not helpful.